### DISCUSSION TITLE
Programatically changing config (Server.xml)

### DISCUSSION THREAD

**Discussion by flaeri** [OvenMediaLabs/OvenMediaEngine#1335] [Ideas]
Hello :) I've been a very happy user of OME for about a year now.
We have a bit of an odd way of using it, but it works surprisingly well.

Essentially we have 10-50 ingests that we make accessible via the <Origins> in the Server.xml file. Example:
```
<Origin>
	<Location>/rtsp/r51front</Location>
	<Pass>
		<Scheme>rtsp</Scheme>
		<Urls><Url>10.xx.xx.x/front/</Url></Urls>
	</Pass>
</Origin>
```
As far as I know, the only way of adding a new Origin is to modify the XML file by hand, so I have about 20 of these, and regularly have to add more.

It would be lovely if there was some way to programatically add (and even remove or change) Origin entries.
The REST API seems like a natural fit for doing so. 

I know this is very niche, but I can see how others could also benefit from a similar way to modify origins, or other parts of the config programatically (like perhaps adding MPEGTS streammap or similar).

In the meantime, I think i'll just have to write an XML parser and writer, but not exactly a clean solution, heh.
Thank you all very much for your hard work and effort <3

**Comment by dimiden** [comment]
@flaeri
Thank you for using OvenMediaEngine!
OvenMediaEngine provides the [OriginMapStore](https://airensoft.gitbook.io/ovenmediaengine/origin-edge-clustering#originmapstore) feature to allow for more flexible configurations.
Please refer to the documentation to set up a Redis server and make the necessary configuration changes.

**Comment by AdamDudley** [comment]
Hi, I'm confused. I thought this was exactly what the API was doing. Adding a new origin? Please let me know if I've misunderstood, as that would explain my current problems...

Create stream
https://airensoft.gitbook.io/ovenmediaengine/rest-api/v1/virtualhost/application/stream

**Reply by getroot** [reply]
Using OriginMapStore, you can add origin stream information by entering data into Redis. Of course, it is also possible through API.
